### PR TITLE
fix(component): don't pass rowdata to cell title

### PR DESCRIPTION
**Pull request by romainseb**
**What is the problem this PR is trying to solve?**
RowData can contain children prop 🙄 
**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by Nigui — APPROVED** [review]

File: .changeset/pink-gorillas-hug.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@talend/react-components': patch
```
Comment (Benrajalu) [inline .changeset/pink-gorillas-hug.md:2]: This is an API break though, with possible breaking side-effects. You guys are sure?

Comment (Benrajalu) [reply .changeset/pink-gorillas-hug.md:2]: Sure, I can believe it's not necessary. But if it has been done somewhere, then this PR becomes breaking, not patch.

Comment (romainseb) [reply .changeset/pink-gorillas-hug.md:2]: I'm confident that this property is not used.
Everything already need is passed aside, spreading all the row data to the "title" button has no usage in our app afaik.

Thanks for raising this point.

Comment (jsomsanith-tlnd) [reply .changeset/pink-gorillas-hug.md:2]: Yes, this was an error at first.
The action represents the title of the tile that is restricted. This idea is not to open it to customisation.
And here we spread all the list's item properties as props, which doesn't make sense. In practice we end up with tons of unexpected props in Action component input, that won't be used.

**Review by jsomsanith-tlnd — APPROVED** [review]

File: .changeset/pink-gorillas-hug.md (hunk 1 above)
Comment (Nigui) [reply .changeset/pink-gorillas-hug.md:2]: I found a side-effect : it removes the `name` attribute of CellTitle ActionButton button tag leading to break selectors.

**Pull request merged**